src/app/shared/example-viewer/example-viewer.spec.ts

```diff
@@ -48,12 +48,57 @@ describe('ExampleViewer', () => {
 
   it('should toggle between the 3 views', async(() => {
     fixture.detectChanges();
-    component.view = 'compact';
-    expect(component.view).toBe('compact');
+    // need to specify a file because toggling from snippet to full changes the tabs to match
+    component.file = 'file.html';
+    component.view = 'snippet';
+    expect(component.view).toBe('snippet');
     component.toggleCompactView();
     expect(component.view).toBe('full');
     component.toggleSourceView();
-    expect(component.view).toBe('collapsed');
+    expect(component.view).toBe('demo');
+  }));
+
+  it('should change to correct tab', async(() => {
+    fixture.detectChanges();
+    component.file = 'file.html';
+    component.selectCorrectTab();
+    expect(component.selectedTab).toBe(0);
+
+    component.file = 'file.ts';
+    component.selectCorrectTab();
+    expect(component.selectedTab).toBe(1);
+
+    component.file = 'file.css';
+    component.selectCorrectTab();
+    expect(component.selectedTab).toBe(2);
+  }));
+
+  it('should generate correct url with region', async(() => {
+    fixture.detectChanges();
+    component.example = exampleKey;
+    component.region = 'region';
+    const url = component.generateUrl('a.b.html');
+    expect(url).toBe(exampleBasePath + '/a.b_region-html.html');
+  }));
+
+  it('should generate correct url without region', async(() => {
+    fixture.detectChanges();
+    component.example = exampleKey;
+    component.region = undefined;
+    const url = component.generateUrl('a.b.ts');
+    expect(url).toBe(exampleBasePath + '/a.b-ts.html');
+  }));
+
+  it('should print an error message about incorrect file type', async(() => {
+    spyOn(console, 'error');
+    fixture.detectChanges();
+    component.file = 'file.bad';
+    component.selectCorrectTab();
+
+
+    expect(console.error).toHaveBeenCalled();
```

Comment:
nit: you can remove this. It's covered by the line below. Error would be 
`expected console.error to have been called with xxxx, but was never called`